pkg/routeagent_driver/constants/constants.go

```diff
@@ -54,5 +54,5 @@ const (
 	NetworkPluginWeaveNet      = "weave-net"
 	NetworkPluginOpenShiftSDN  = "OpenShiftSDN"
 	NetworkPluginOVNKubernetes = "OVNKubernetes"
-	NetworkPluginCalico        = "calico"
+	NetworkPluginCalico        = "Calico"
```

Comment:
In non-OpenShift setups, the plugin is named `calico`; we can’t just replace the name here.

Reply:
Thanks @skitt, 
Hmm, how bout adding a new entry to the constants list (something like OcpNetworkPluginCalico = "Calico" )?

Reply:
Rather than adding a new constant, how about doing a case insensitive comparison at the place where its validated?

Reply:
@yboaron If you decide not to change the `NetworkPluginCalico` constant and do a case insensitive comparison, make sure to also change the comparison in:
https://github.com/submariner-io/submariner-operator/blob/5eb19252606e888f444bcb79ccf78c3ec4331105/pkg/subctl/cmd/diagnose/cni.go#L73

Reply:
Thanks for the feedback @dberg1 ,
I'm new to Submariner, maybe I'm missing something here ...

So, the operator is running auto discovery process and updates 
submariner.Status.NetworkPlugin with clusterNetwork.NetworkPlugin [1]
and for Calico the clusterNetwork.NetworkPlugin is set to constant from submariner code [2]

It isn't clear to me how we can have a mismatch in plugin name in that case, 
could you please elaborate on how you hit the case described in [3] ?

[1]
https://github.com/submariner-io/submariner-operator/blob/devel/controllers/submariner/submariner_networkdiscovery.go#L66
[2]
https://github.com/submariner-io/submariner-operator/blob/devel/pkg/discovery/network/calico.go#L55
[3]
https://github.com/submariner-io/submariner/issues/1652

Reply:
@yboaron `discoverNetwork` returns with `err != nil` since Calico does not match any supported plugin:
https://github.com/submariner-io/submariner-operator/blob/2287e445e48bc1ebcc7f9648ec5f80eed3af9ea5/controllers/submariner/submariner_networkdiscovery.go#L51
so it returns here: 
https://github.com/submariner-io/submariner-operator/blob/2287e445e48bc1ebcc7f9648ec5f80eed3af9ea5/pkg/discovery/network/calico.go#L51

Reply:
Not sure that I fully understood your comment, and I don't see why the issue you described above is specific to calico/Calico.

From what I understand,  discoverNetwork will return error [1] for all the plugins for the first time ( before r.config.ClusterNetwork was cached).

How about adding a new line below [2] : 
" return return r.config.ClusterNetwork, nil "     ?

[1] https://github.com/submariner-io/submariner-operator/blob/2287e445e48bc1ebcc7f9648ec5f80eed3af9ea5/controllers/submariner/submariner_networkdiscovery.go#L36-#L51

[2] 
https://github.com/submariner-io/submariner-operator/blob/2287e445e48bc1ebcc7f9648ec5f80eed3af9ea5/controllers/submariner/submariner_networkdiscovery.go#L45

Reply:
@dberg1  If you can still access your Ocp cluster with calico it will be useful if you can share the following:
- NetworkPlugin env var from Routeagent daemonset manifest [1]
- logs from one of submariner-routeagent pods  [2]

it may hint at the source of the problem


[1] kubectl get ds -n submariner-operator  submariner-routeagent  -o yaml  | grep -A 2 NETWORKPLU

[2] kubectl logs  -n submariner-operator submariner-routeagent-j8jf4

Reply:
https://github.com/submariner-io/submariner/issues/1652#issuecomment-1020248034

Reply:
Please check https://github.com/submariner-io/submariner/issues/1652#issuecomment-1020356088

Reply:
Following the discussion in https://github.com/submariner-io/submariner/issues/1652 , I think we should address this issue in the network-discovery code in the operator.

I'm going to close this PR and create a new PR for the operator repo